File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-2.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0;
        int r = nums.length-1;
        int mid;
        while (l < r - 1) {
            mid = (l + r)/2;
            System.out.println(r);
            if (target == nums[l]) {
                return l;
            }
            if (target == nums[r]) {
                return r;
            }
            if (target < nums[mid]) {
                if (nums[l] < target) {
                    r = mid;
                } else {
                    if (nums[l] < nums[mid]) {
                        l = mid;
                    } else {
                        r = mid;
                    }
                }
            } else {
                if (target < nums[r]) {
                    l = mid;
                } else {
                    if (nums[mid] < nums[r]) {
                        r = mid;
                    } else {
                        l = mid;
                    }
                }
            }
        }
        if (target == nums[l]) {
            return l;
        }
        if (target == nums[r]) {
            return r;
        }
        return -1;
    }
}

File: Data Structures & Algorithms/binary-search/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1
        while (l < r):
            mid = l + (r-l)//2
            print(f"l{l}, r{r}, mid{mid}")
            if target < nums[mid]:
                r = mid - 1
            elif target > nums[mid]:
                l = mid + 1
            else:
                return mid
        if (nums[l] == target):
            return l
        else:
            return -1